File: 1752-check-if-array-is-sorted-and-rotated/1752-check-if-array-is-sorted-and-rotated.py
class Solution:
    def check(self, nums: List[int]) -> bool:
        count_decreasing = 0
        for i in range(0, len(nums)):
            if nums[i] > nums[(i + 1) % len(nums)]:
                count_decreasing += 1
            
            if count_decreasing > 1:
                return False
        
        return True

# array
# time: O(n)
# space: O(1)

File: 1752-check-if-array-is-sorted-and-rotated/1752-check-if-array-is-sorted-and-rotated.go
func check(nums []int) bool {
	sorted := true
	count := 0
	for i := 1; i < len(nums); i++ {
		if nums[i] < nums[i-1] {
			sorted = false
			count++
		}
	}

	if sorted {
		return true
	} else {
		if count > 1 {
			return false
		} else {
			if nums[0] >= nums[len(nums)-1] {
				return true
			} else {
				return false
			}
		}
	}
}

// array
// time: O(n)
// space: O(1)

// #1 2025/02/02
// #2 2026/05/23